test/components/routers/test_conditional_router.py

```diff
@@ -542,3 +542,29 @@ def test_warns_on_unused_optional_variables(self):
         # Verify router still works normally
         result = router.run(question="What?", mode="chat")
         assert result == {"chat": "What?"}
+
+    def test_router_to_dict_does_not_mutate_routes(self):
+        routes = [
```

Comment:
I would add one more  "roundtrip" test relying on `routes`

initializing a new ConditionalRouter from `router_dict` and asserting that the routes are the same.